DirectProgramming/C++SYCL_FPGA/Tutorials/GettingStarted/fpga_compile/part5-dpcpp_functor_buffers/src/vector_add.cpp

```diff
@@ -0,0 +1,116 @@
+#include <iostream>
+
+// oneAPI headers
+#include <CL/sycl.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+
+using namespace sycl;
+
+// Forward declare the kernel name in the global scope. This is an FPGA best
+// practice that reduces name mangling in the optimization reports.
+class VectorAddID;
+
+template <class AccA, class AccB, class AccC>
```

Comment:
Declaring the struct with template type arguments (then using declttype) works, but does not do the same thing the other examples do.   The challenge is the original arrays needs to be read-write.   But you are capturing a read-only reference in the a struct.    The code "works", but if the intention is to show that equivalence, than it does not accomplish that.

Reply:
OK, removing part5 to keep consistency across